crates/autopilot/src/infra/mod.rs

```diff
@@ -3,4 +3,9 @@ pub mod persistence;
 pub mod shadow;
 pub mod solvers;
 
-pub use {blockchain::Ethereum, persistence::Persistence, solvers::Driver};
+pub use {
+    blockchain::Ethereum,
+    order_validation::banned,
```

Comment:
I wouldn't import this as a crate. I would put the `ChainalysisOracle` inside autopilot `blockchain::Ethereum::Contracts` and have the actual implementation under `infra::Ethereum`

Reply:
If there is a possibility to extract something into a separate crate without violating constraints we want to follow then I would do so, otherwise I would accept code duplication.

Btw, we should be able to instantiate the contract inside infra::blockchain::contracts and forward it to the `infra::banned::Users::new` constructor instead of web3.

Reply:
What is your concern with importing this crate? That the interface is not using the autopilot specific types (e.g. eth::Address vs H160) or something else? What constraints are violated?

I really don't want to duplicate the logic of the banned implementation (the way we do the caching is non-trivial) and instead suggest to add a shim to the autopilot infra that translates between the separate module and autopilot itself. However, I'd like to understand how that interface would be different to the current one.

> Btw, we should be able to instantiate the contract inside infra::blockchain::contracts

That struct only contains addresses (we need the instance so we can invoke a function on it) and also seems to be mainly for core CoW Protocol relevant contracts that are expected to exist on all chains (here we only support mainnet).

Reply:
No real concern other than the consistency. Web3 services should only be accessed through `infra::Ethereum` interface. All contracts should be in one place to be instantiated once, potentially reused, and accessed through `infra::Ethereum` interface. At least I thought this is the way we want to decouple things around in autopilot.
But I get the duplication is pain.

Reply:
I moved the contracts on the contracts struct and pass it into the banned::User constructor. 

Let me know if you feel `infra::blockchain` should implement its own banned::User component (note that the current implementation is available via `infra::banned::Users`) but please also help me understand the difference in the APIs (I wouldn't want to create a shim that implements the exact same API).